sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1432,4 +1434,24 @@ object HiveExternalCatalog {
       isHiveCompatibleDataType(m.keyType) && isHiveCompatibleDataType(m.valueType)
     case _ => true
   }
+
+  /**
+   * Rewrite uri to absolute location. For example:
+   *   uri: /user/hive/warehouse/test_table
+   *   parentUri: viewfs://clusterA/user/hive/warehouse/
+   *   The result is: viewfs://clusterA/user/hive/warehouse/test_table
+   */
+  private[spark] def toAbsoluteURI(uri: URI, parentUri: Option[URI]): URI = {
```

Comment:
I think the name `parentUri` is a bit misleading. It's more like an `absoluteUri` which we need to inherit its schema, host, port, etc.

Reply:
Renamed it to absoluteUri.